### DISCUSSION TITLE
docker build --platform linux/amd64 -t lemmy-ui . 195 error code ERESOLVE 196 error ERESOLVE unable to resolve dependency tree

### DISCUSSION THREAD

**Discussion by qiancsff** [LemmyNet/lemmy-ui#2616] [Q&A]
root@debian:~/lemmy-ui# node -v # should print `v20.15.1`
npm -v # should print `10.7.0`
v20.15.1
10.7.0
root@debian:~/lemmy-ui# npm install
npm error code ERESOLVE
npm error ERESOLVE unable to resolve dependency tree
npm error
npm error While resolving: lemmy-ui@0.19.5
npm error Found: eslint@9.7.0
npm error node_modules/eslint
npm error   dev eslint@"^9.4.0" from the root project
npm error
npm error Could not resolve dependency:
npm error peer eslint@"^8.56.0" from @typescript-eslint/parser@7.16.1
npm error node_modules/@typescript-eslint/parser
npm error   dev @typescript-eslint/parser@"^7.12.0" from the root project
npm error   peer @typescript-eslint/parser@"^7.0.0" from @typescript-eslint/eslint-plugin@7.16.1
npm error   node_modules/@typescript-eslint/eslint-plugin
npm error     dev @typescript-eslint/eslint-plugin@"^7.12.0" from the root project
npm error
npm error Fix the upstream dependency conflict, or retry
npm error this command with --force or --legacy-peer-deps
npm error to accept an incorrect (and potentially broken) dependency resolution.
npm error
npm error
npm error For a full report see:
npm error /root/.npm/_logs/2024-07-19T01_10_59_125Z-eresolve-report.txt

npm error A complete log of this run can be found in: /root/.npm/_logs/2024-07-19T01_10_59_125Z-debug-0.log
root@debian:~/lemmy-ui# cat /root/.npm/_logs/2024-07-19T01_10_59_125Z-debug-0.log
0 verbose cli /root/.local/share/fnm/node-versions/v20.15.1/installation/bin/node /run/user/0/fnm_multishells/25063_1721351377949/bin/npm
1 info using npm@10.7.0
2 info using node@v20.15.1
3 silly config:load:file:/root/.local/share/fnm/node-versions/v20.15.1/installation/lib/node_modules/npm/npmrc
4 silly config:load:file:/root/lemmy-ui/.npmrc
5 silly config:load:file:/root/.npmrc
6 silly config:load:file:/root/.local/share/fnm/node-versions/v20.15.1/installation/etc/npmrc
7 verbose title npm install
8 verbose argv "install"
9 verbose logfile logs-max:10 dir:/root/.npm/_logs/2024-07-19T01_10_59_125Z-
10 verbose logfile /root/.npm/_logs/2024-07-19T01_10_59_125Z-debug-0.log
11 silly logfile done cleaning log files
12 silly idealTree buildDeps
13 silly fetch manifest @babel/plugin-proposal-decorators@^7.24.7
14 http fetch GET 200 https://registry.npmjs.org/@babel%2fplugin-proposal-decorators 167ms (cache miss)
15 silly fetch manifest @babel/core@^7.24.7
16 http fetch GET 200 https://registry.npmjs.org/@babel%2fcore 62ms (cache miss)
17 silly fetch manifest @babel/plugin-transform-class-properties@^7.24.7
18 http fetch GET 200 https://registry.npmjs.org/@babel%2fplugin-transform-class-properties 41ms (cache miss)
19 silly fetch manifest @babel/plugin-transform-runtime@^7.24.7
20 http fetch GET 200 https://registry.npmjs.org/@babel%2fplugin-transform-runtime 56ms (cache miss)
21 silly fetch manifest @babel/plugin-transform-typescript@^7.24.7
22 http fetch GET 200 https://registry.npmjs.org/@babel%2fplugin-transform-typescript 49ms (cache miss)
23 silly fetch manifest @babel/preset-env@^7.24.7
24 http fetch GET 200 https://registry.npmjs.org/@babel%2fpreset-env 72ms (cache miss)
25 silly fetch manifest @babel/preset-typescript@^7.24.7
26 http fetch GET 200 https://registry.npmjs.org/@babel%2fpreset-typescript 68ms (cache miss)
27 silly fetch manifest @babel/runtime@^7.24.7
28 http fetch GET 200 https://registry.npmjs.org/@babel%2fruntime 99ms (cache miss)
29 silly fetch manifest @emoji-mart/data@^1.2.1
30 http fetch GET 200 https://registry.npmjs.org/@emoji-mart%2fdata 114ms (cache miss)
31 silly fetch manifest autosize@^6.0.1
32 http fetch GET 200 https://registry.npmjs.org/autosize 606ms (cache miss)
33 silly fetch manifest babel-loader@^9.1.3
34 http fetch GET 200 https://registry.npmjs.org/babel-loader 40ms (cache miss)
35 silly fetch manifest webpack@^5.91.0
36 http fetch GET 200 https://registry.npmjs.org/webpack 150ms (cache miss)
37 silly fetch manifest babel-plugin-inferno@^6.7.1
38 http fetch GET 200 https://registry.npmjs.org/babel-plugin-inferno 861ms (cache miss)
39 silly fetch manifest bootstrap@^5.3.3
40 http fetch GET 200 https://registry.npmjs.org/bootstrap 59ms (cache miss)
41 silly fetch manifest @popperjs/core@^2.11.8
42 http fetch GET 200 https://registry.npmjs.org/@popperjs%2fcore 64ms (cache miss)
43 silly fetch manifest check-password-strength@^2.0.10
44 http fetch GET 200 https://registry.npmjs.org/check-password-strength 629ms (cache miss)
45 silly fetch manifest classnames@^2.5.1
46 http fetch GET 200 https://registry.npmjs.org/classnames 31ms (cache miss)
47 silly fetch manifest clean-webpack-plugin@^4.0.0
48 http fetch GET 200 https://registry.npmjs.org/clean-webpack-plugin 39ms (cache miss)
49 silly fetch manifest cookie@^0.6.0
50 http fetch GET 200 https://registry.npmjs.org/cookie 36ms (cache miss)
51 silly fetch manifest cookie-parser@^1.4.6
52 http fetch GET 200 https://registry.npmjs.org/cookie-parser 33ms (cache miss)
53 silly fetch manifest copy-webpack-plugin@^12.0.2
54 http fetch GET 200 https://registry.npmjs.org/copy-webpack-plugin 37ms (cache miss)
55 silly fetch manifest css-loader@^7.1.2
56 http fetch GET 200 https://registry.npmjs.org/css-loader 52ms (cache miss)
57 silly fetch manifest @rspack/core@0.x || 1.x
58 http fetch GET 200 https://registry.npmjs.org/@rspack%2fcore 161ms (cache miss)
59 silly fetch manifest @swc/helpers@>=0.5.1
60 http fetch GET 200 https://registry.npmjs.org/@swc%2fhelpers 84ms (cache miss)
61 silly fetch manifest date-fns@^3.6.0
62 http fetch GET 200 https://registry.npmjs.org/date-fns 167ms (cache miss)
63 silly fetch manifest emoji-mart@^5.6.0
64 http fetch GET 200 https://registry.npmjs.org/emoji-mart 62ms (cache miss)
65 silly fetch manifest emoji-short-name@^2.0.0
66 http fetch GET 200 https://registry.npmjs.org/emoji-short-name 640ms (cache miss)
67 silly fetch manifest express@~4.19.2
68 http fetch GET 200 https://registry.npmjs.org/express 55ms (cache miss)
69 silly fetch manifest highlight.js@^11.9.0
70 http fetch GET 200 https://registry.npmjs.org/highlight.js 76ms (cache miss)
71 silly fetch manifest history@^5.3.0
72 http fetch GET 200 https://registry.npmjs.org/history 92ms (cache miss)
73 silly fetch manifest html-to-text@^9.0.5
74 http fetch GET 200 https://registry.npmjs.org/html-to-text 33ms (cache miss)
75 silly fetch manifest i18next@^23.11.5
76 http fetch GET 200 https://registry.npmjs.org/i18next 72ms (cache miss)
77 silly fetch manifest inferno@^8.2.3
78 http fetch GET 200 https://registry.npmjs.org/inferno 680ms (cache miss)
79 silly fetch manifest inferno-create-element@^8.2.3
80 http fetch GET 200 https://registry.npmjs.org/inferno-create-element 617ms (cache miss)
81 silly fetch manifest inferno-helmet@^5.2.1
82 http fetch GET 200 https://registry.npmjs.org/inferno-helmet 619ms (cache miss)
83 silly fetch manifest inferno-hydrate@^8.2.3
84 http fetch GET 200 https://registry.npmjs.org/inferno-hydrate 237ms (cache miss)
85 silly fetch manifest inferno-i18next-dess@0.0.2
86 http fetch GET 200 https://registry.npmjs.org/inferno-i18next-dess 631ms (cache miss)
87 silly fetch manifest inferno-router@^8.2.3
88 http fetch GET 200 https://registry.npmjs.org/inferno-router 1517ms (cache miss)
89 silly fetch manifest inferno-server@^8.2.3
90 http fetch GET 200 https://registry.npmjs.org/inferno-server 646ms (cache miss)
91 silly fetch manifest jwt-decode@^4.0.0
92 http fetch GET 200 https://registry.npmjs.org/jwt-decode 32ms (cache miss)
93 silly fetch manifest lemmy-js-client@0.19.4
94 http fetch GET 200 https://registry.npmjs.org/lemmy-js-client 1878ms (cache miss)
95 silly fetch manifest lodash.isequal@^4.5.0
96 http fetch GET 200 https://registry.npmjs.org/lodash.isequal 29ms (cache miss)
97 silly fetch manifest markdown-it@^14.1.0
98 http fetch GET 200 https://registry.npmjs.org/markdown-it 55ms (cache miss)
99 silly fetch manifest markdown-it-bidi@^0.1.0
100 http fetch GET 200 https://registry.npmjs.org/markdown-it-bidi 618ms (cache miss)
101 silly fetch manifest markdown-it-container@^4.0.0
102 http fetch GET 200 https://registry.npmjs.org/markdown-it-container 590ms (cache miss)
103 silly fetch manifest markdown-it-emoji@^3.0.0
104 http fetch GET 200 https://registry.npmjs.org/markdown-it-emoji 34ms (cache miss)
105 silly fetch manifest markdown-it-footnote@^4.0.0
106 http fetch GET 200 https://registry.npmjs.org/markdown-it-footnote 602ms (cache miss)
107 silly fetch manifest markdown-it-highlightjs@^4.1.0
108 http fetch GET 200 https://registry.npmjs.org/markdown-it-highlightjs 599ms (cache miss)
109 silly fetch manifest markdown-it-html5-embed@^1.0.0
110 http fetch GET 200 https://registry.npmjs.org/markdown-it-html5-embed 603ms (cache miss)
111 silly fetch manifest markdown-it-ruby@^0.1.1
112 http fetch GET 200 https://registry.npmjs.org/markdown-it-ruby 598ms (cache miss)
113 silly fetch manifest markdown-it-sub@^2.0.0
114 http fetch GET 200 https://registry.npmjs.org/markdown-it-sub 630ms (cache miss)
115 silly fetch manifest markdown-it-sup@^2.0.0
116 http fetch GET 200 https://registry.npmjs.org/markdown-it-sup 595ms (cache miss)
117 silly fetch manifest mini-css-extract-plugin@^2.9.0
118 http fetch GET 200 https://registry.npmjs.org/mini-css-extract-plugin 55ms (cache miss)
119 silly fetch manifest qreator@^9.2.0
120 http fetch GET 200 https://registry.npmjs.org/qreator 603ms (cache miss)
121 silly fetch manifest register-service-worker@^1.7.2
122 http fetch GET 200 https://registry.npmjs.org/register-service-worker 31ms (cache miss)
123 silly fetch manifest run-node-webpack-plugin@^1.3.0
124 http fetch GET 200 https://registry.npmjs.org/run-node-webpack-plugin 593ms (cache miss)
125 silly fetch manifest rxjs@^7.8.1
126 http fetch GET 200 https://registry.npmjs.org/rxjs 62ms (cache miss)
127 silly fetch manifest sanitize-html@^2.13.0
128 http fetch GET 200 https://registry.npmjs.org/sanitize-html 59ms (cache miss)
129 silly fetch manifest sass@^1.77.4
130 http fetch GET 200 https://registry.npmjs.org/sass 73ms (cache miss)
131 silly fetch manifest sass-loader@^14.2.1
132 http fetch GET 200 https://registry.npmjs.org/sass-loader 42ms (cache miss)
133 silly fetch manifest node-sass@^4.0.0 || ^5.0.0 || ^6.0.0 || ^7.0.0 || ^8.0.0 || ^9.0.0
134 http fetch GET 200 https://registry.npmjs.org/node-sass 99ms (cache miss)
135 silly fetch manifest sass-embedded@*
136 http fetch GET 200 https://registry.npmjs.org/sass-embedded 37ms (cache miss)
137 silly fetch manifest serialize-javascript@^6.0.2
138 http fetch GET 200 https://registry.npmjs.org/serialize-javascript 34ms (cache miss)
139 silly fetch manifest service-worker-webpack@^1.0.0
140 http fetch GET 200 https://registry.npmjs.org/service-worker-webpack 226ms (cache miss)
141 silly fetch manifest sharp@0.33.4
142 http fetch GET 200 https://registry.npmjs.org/sharp 44ms (cache miss)
143 silly fetch manifest tippy.js@^6.3.7
144 http fetch GET 200 https://registry.npmjs.org/tippy.js 46ms (cache miss)
145 silly fetch manifest toastify-js@^1.12.0
146 http fetch GET 200 https://registry.npmjs.org/toastify-js 44ms (cache miss)
147 silly fetch manifest tributejs@^5.1.3
148 http fetch GET 200 https://registry.npmjs.org/tributejs 624ms (cache miss)
149 silly fetch manifest webpack-cli@^5.1.4
150 http fetch GET 200 https://registry.npmjs.org/webpack-cli 50ms (cache miss)
151 silly fetch manifest webpack-node-externals@^3.0.0
152 http fetch GET 200 https://registry.npmjs.org/webpack-node-externals 37ms (cache miss)
153 silly fetch manifest @eslint/js@^9.4.0
154 http fetch GET 200 https://registry.npmjs.org/@eslint%2fjs 55ms (cache miss)
155 silly fetch manifest @types/autosize@^4.0.3
156 http fetch GET 200 https://registry.npmjs.org/@types%2fautosize 1176ms (cache miss)
157 silly fetch manifest @types/bootstrap@^5.2.10
158 http fetch GET 200 https://registry.npmjs.org/@types%2fbootstrap 128ms (cache miss)
159 silly fetch manifest @types/cookie@^0.6.0
160 http fetch GET 200 https://registry.npmjs.org/@types%2fcookie 39ms (cache miss)
161 silly fetch manifest @types/cookie-parser@^1.4.7
162 http fetch GET 200 https://registry.npmjs.org/@types%2fcookie-parser 826ms (cache miss)
163 silly fetch manifest @types/express@^4.17.21
164 http fetch GET 200 https://registry.npmjs.org/@types%2fexpress 42ms (cache miss)
165 silly fetch manifest @types/html-to-text@^9.0.4
166 http fetch GET 200 https://registry.npmjs.org/@types%2fhtml-to-text 45ms (cache miss)
167 silly fetch manifest @types/lodash.isequal@^4.5.8
168 http fetch GET 200 https://registry.npmjs.org/@types%2flodash.isequal 1207ms (cache miss)
169 silly fetch manifest @types/markdown-it@^14.1.1
170 http fetch GET 200 https://registry.npmjs.org/@types%2fmarkdown-it 50ms (cache miss)
171 silly fetch manifest @types/markdown-it-container@^2.0.10
172 http fetch GET 200 https://registry.npmjs.org/@types%2fmarkdown-it-container 65ms (cache miss)
173 silly fetch manifest @types/node@^20.14.2
174 http fetch GET 200 https://registry.npmjs.org/@types%2fnode 214ms (cache miss)
175 silly fetch manifest @types/path-browserify@^1.0.2
176 http fetch GET 200 https://registry.npmjs.org/@types%2fpath-browserify 823ms (cache miss)
177 silly fetch manifest @types/sanitize-html@^2.11.0
178 http fetch GET 200 https://registry.npmjs.org/@types%2fsanitize-html 45ms (cache miss)
179 silly fetch manifest @types/serialize-javascript@^5.0.4
180 http fetch GET 200 https://registry.npmjs.org/@types%2fserialize-javascript 36ms (cache miss)
181 silly fetch manifest @types/toastify-js@^1.12.3
182 http fetch GET 200 https://registry.npmjs.org/@types%2ftoastify-js 1221ms (cache miss)
183 silly fetch manifest @typescript-eslint/eslint-plugin@^7.12.0
184 http fetch GET 200 https://registry.npmjs.org/@typescript-eslint%2feslint-plugin 301ms (cache miss)
185 silly fetch manifest @typescript-eslint/parser@^7.12.0
186 http fetch GET 200 https://registry.npmjs.org/@typescript-eslint%2fparser 354ms (cache miss)
187 silly fetch manifest eslint@^9.4.0
188 http fetch GET 200 https://registry.npmjs.org/eslint 50ms (cache miss)
189 silly fetch manifest eslint@^8.56.0
190 verbose stack Error: unable to resolve dependency tree
190 verbose stack     at #failPeerConflict (/root/.local/share/fnm/node-versions/v20.15.1/installation/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/build-ideal-tree.js:1386:25)
190 verbose stack     at #loadPeerSet (/root/.local/share/fnm/node-versions/v20.15.1/installation/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/build-ideal-tree.js:1352:33)
190 verbose stack     at async #loadPeerSet (/root/.local/share/fnm/node-versions/v20.15.1/installation/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/build-ideal-tree.js:1331:23)
190 verbose stack     at async #buildDepStep (/root/.local/share/fnm/node-versions/v20.15.1/installation/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/build-ideal-tree.js:923:11)
190 verbose stack     at async Arborist.buildIdealTree (/root/.local/share/fnm/node-versions/v20.15.1/installation/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/build-ideal-tree.js:203:7)
190 verbose stack     at async Promise.all (index 1)
190 verbose stack     at async Arborist.reify (/root/.local/share/fnm/node-versions/v20.15.1/installation/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/reify.js:131:5)
190 verbose stack     at async Install.exec (/root/.local/share/fnm/node-versions/v20.15.1/installation/lib/node_modules/npm/lib/commands/install.js:150:5)
190 verbose stack     at async module.exports (/root/.local/share/fnm/node-versions/v20.15.1/installation/lib/node_modules/npm/lib/cli/entry.js:74:5)
191 verbose cwd /root/lemmy-ui
192 verbose Linux 5.10.0-15-amd64
193 verbose node v20.15.1
194 verbose npm  v10.7.0
195 error code ERESOLVE
196 error ERESOLVE unable to resolve dependency tree
197 error
198 error While resolving: lemmy-ui@0.19.5
198 error Found: eslint@9.7.0
198 error node_modules/eslint
198 error   dev eslint@"^9.4.0" from the root project
198 error
198 error Could not resolve dependency:
198 error peer eslint@"^8.56.0" from @typescript-eslint/parser@7.16.1
198 error node_modules/@typescript-eslint/parser
198 error   dev @typescript-eslint/parser@"^7.12.0" from the root project
198 error   peer @typescript-eslint/parser@"^7.0.0" from @typescript-eslint/eslint-plugin@7.16.1
198 error   node_modules/@typescript-eslint/eslint-plugin
198 error     dev @typescript-eslint/eslint-plugin@"^7.12.0" from the root project
198 error
198 error Fix the upstream dependency conflict, or retry
198 error this command with --force or --legacy-peer-deps
198 error to accept an incorrect (and potentially broken) dependency resolution.
199 error
199 error
199 error For a full report see:
199 error /root/.npm/_logs/2024-07-19T01_10_59_125Z-eresolve-report.txt
200 verbose exit 1
201 verbose code 1
202 silly unfinished npm timer reify 1721351459749
203 silly unfinished npm timer reify:loadTrees 1721351459755
204 silly unfinished npm timer idealTree:buildDeps 1721351459780
205 silly unfinished npm timer idealTree:#root 1721351459781

**Comment by qiancsff** [comment]
npm install eslint@^8.56.0
After installation,
Docker build crashes

**Reply by dessalines** [reply]
Use `pnpm install`
